### DISCUSSION TITLE
Channel read api is missing frameworks

### DISCUSSION THREAD

**Discussion by pranavirabatti** [Sunbird-Knowlg/Community#18] [Q&A] [answered]
Suggested frameworks or frameworks? In the latest version, I see only frameworks. In the code, they are trying to access frameworks and not suggested_frameworks. 

API: domain_name/api/channel/v1/read/channel_id

On sunbird portal version: 4.4.1

![image](https://user-images.githubusercontent.com/22250006/173382793-f0b7bf0c-551b-404f-8d70-eb7a358e061d.png)

On latest sunbird: 

![image](https://user-images.githubusercontent.com/22250006/173383518-2ac272e3-85d4-4b4a-804e-9acd5fa84f27.png)

@vinukumar-vs @rajeevsathish

**Comment by AmiableAnil** [comment, marked as answer]
@pranavirabatti Read channel API returns **_suggested_framworks_** only if _**frameworks**_ is not available. It does not return both at the same time. Please find it [here](https://github.com/project-sunbird/sunbird-learning-platform/blob/8a6a67f78c76f0e3107b1a1ea95c0cb8d5a8171b/platform-modules/sunbird-framework/src/main/java/org/sunbird/framework/mgr/impl/ChannelManagerImpl.java#L68).

**Reply by pranavirabatti** [reply]
Do you mean to say that frameworks are not yet mapped at the channel level in our case? If it is mapped then it will show the frameworks else suggested frameworks. Correct me If I'm wrong.

**Reply by vinukumar-vs** [reply]
Yes. None of the frameworks are mapped to the channel, hence it is taking the suggested_frameworks.

**Reply by pranavirabatti** [reply]
@AmiableAnil @vinukumar-vs Thanks!
